web/src/EducationBenchmarking.Web/Controllers/SchoolPlanningYearController.cs

```diff
@@ -25,14 +26,22 @@ public async Task<IActionResult> Index(string urn, int year)
         {
             try
             {
-                //TODO: Get if exists plan for school / year
-                //TODO: Display previous selection value if plan exists
                 ViewData[ViewDataConstants.Backlink] = new BacklinkInfo("SelectYear", "SchoolPlanning", new { urn });
 
                 var school = await establishmentApi.GetSchool(urn).GetResultOrThrow<School>();
                 var finances = await financeService.GetFinances(school);
-                var viewModel = new SchoolPlanViewModel(school, finances, year);
-                return View(viewModel);
+                
+                try
+                {
+                    var existingPlan = await benchmarkApi.GetFinancialPlan(urn, year).GetResultOrThrow<FinancialPlanResult>();
```

Comment:
Throwing an exception for this is expensive. Consider the `GetResultOrDefault` ApiResult extension method

Reply:
fixed - amended in latest commit